packages/contracts-bedrock/scripts/upgrades/holocene/DeployUpgrade.s.sol

```diff
@@ -0,0 +1,333 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Forge
+import { console2 as console } from "forge-std/console2.sol";
+
+// Scripts
+import { Deployer } from "scripts/deploy/Deployer.sol";
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+
+// Utils
+import "src/dispute/lib/Types.sol";
+
+// Interfaces
+import { ISystemConfig } from "src/L1/interfaces/ISystemConfig.sol";
+import { ISuperchainConfig } from "src/L1/interfaces/ISuperchainConfig.sol";
+import { IProxy } from "src/universal/interfaces/IProxy.sol";
+import {
+    IFaultDisputeGame,
+    IBigStepper,
+    IAnchorStateRegistry,
+    IDelayedWETH
+} from "src/dispute/interfaces/IFaultDisputeGame.sol";
+import { IPermissionedDisputeGame } from "src/dispute/interfaces/IPermissionedDisputeGame.sol";
+import { IMIPS, IPreimageOracle } from "src/cannon/interfaces/IMIPS.sol";
+
+/// @title DeployUpgrade
+/// @notice A deployment script for smart contract upgrades surrounding the Holocene hardfork.
+contract DeployUpgrade is Deployer {
+    /// @dev The entrypoint to the deployment script.
+    function deploy(
+        address _proxyAdmin,
+        address _superchainConfig,
+        address _systemConfigImpl,
+        address _mipsImpl,
+        address _delayedWETH,
+        address _preimageOracle,
+        address _anchorStateRegistry,
+        bool _usePermissionlessFaultProofs
+    )
```

Comment:
Seems we could remove a lot of the inputs here by fetching addresses starting from the system config, since from that contract you can find the proxyAdmin, superchainConfig, and existing FP contracts?

Reply:
This already works, plus it's a bit awkward fetching the `ProxyAdmin` + `SuperchainConfig` from the `SystemConfig`. It doesn't have direct references to either. I suppose we could ping the portal to get the `SuperchainConfig` reference and then check the admin of the proxy to get the `ProxyAdmin`, though.